src/components/SearchModal/SearchModal.js

```diff
@@ -215,54 +309,106 @@ const SearchModal = ({
       && Object.keys(otherFilters).length && searchQuery;
 
     if (shouldPerformSearchByTags || shouldPerformRegularSearch || shouldPerformSearchByAccessTypes) {
-      mutator.packages.reset();
-      setLastFetchedPage(1);
+      if (searchType === searchTypes.PACKAGE) {
+        mutator.packages.reset();
+      } else {
+        mutator.titles.reset();
+      }
+
+      changeCurrentSearchConfig(prev => ({
+        ...prev,
+        lastFetchedPage: 1,
+      }));
 
       fetchItems({
         searchQuery,
         searchFilters
       });
     }
-  }, [searchByTagsEnabled, searchAccessTypesEnabled, searchFilters]);
+  }, [
+    currentSearchConfig.searchByTagsEnabled,
+    currentSearchConfig.searchAccessTypesEnabled,
+    currentSearchConfig.searchFilters,
+  ]);
+
+  const onSearchQueryChange = ({ target: { value } }) => {
+    changeCurrentSearchConfig(prev => ({
+      ...prev,
+      searchQuery: value,
+    }));
+  };
 
-  const onSearchQueryChange = e => {
-    setSearchQuery(e.target.value);
+  const onSearchFiltersChange = (filters) => {
```

Comment:
```suggestion
  const onSearchFiltersChange = filters => {
```
If you use one code style, please don't switch to another